src/compiler/binder.ts

```diff
@@ -2627,8 +2627,15 @@ function createBinder(): (file: SourceFile, options: CompilerOptions) => void {
         }
     }
 
+    function checkStrictModeStringLiteral(node: StringLiteral) {
+        if (languageVersion >= ScriptTarget.ES5 && inStrictMode && node.rangesOfOctalSequences) {
+            file.bindDiagnostics.push(...node.rangesOfOctalSequences.map(

```

Comment:
We try to avoid `.push(...thing)` to avoid weird worst-case stack overflow risks.

Reply:
I did find multiple use cases in src by searching `.push(...` – perhaps we should totally get rid of it?